app/backend/src/couchers/servicers/blocking.py

```diff
@@ -1,14 +1,17 @@
 import grpc
 from google.protobuf import empty_pb2
-from sqlalchemy.sql import union
+from sqlalchemy.sql import not_, union
 
 from couchers import errors, urls
 from couchers.models import Upload, User, UserBlock
 from couchers.sql import couchers_select as select
 from proto import blocking_pb2, blocking_pb2_grpc
 
 
-def are_blocked(session, user1_id, user2_id):
+def is_not_visible(session, user1_id, user2_id) -> bool:
+    """
+    Check if user2 is not visible to user1 (due to block from either direction or because account is deleted/banned).
```

Comment:
I would make this function symmetric, just need to add the `where` clause in `hidden_users` as before. Otherwise it's a bit dangerous and you'd need to audit the usages carefully.

Otherwise please make it very explicit with keyword only args:

```python3
def is_not_visible(session, *, this_user=user1_id, other_user=user2_id) -> bool:
```

Reply:
Fair, I guess there is not really a use case where there can be a notification outgoing from an invisible user. Made it symmetric 👍